pkg/server/datastore/sqlstore/sqlstore.go

```diff
@@ -1418,6 +1446,55 @@ func listAttestedNodes(ctx context.Context, db *sqlDB, log logrus.FieldLogger, r
 	}
 }
 
+func createAttestedNodeEvent(tx *gorm.DB, spiffeID string) error {
+	if !fflag.IsSet(fflag.FlagEventsBasedCache) {
+		return nil
+	}
+
+	newAttestedNodeEvent := AttestedNodeEvent{
+		SpiffeID: spiffeID,
+	}
+
+	if err := tx.Create(&newAttestedNodeEvent).Error; err != nil {
+		return sqlError.Wrap(err)
+	}
+
+	return nil
+}
+
+func listAttestedNodesEvents(tx *gorm.DB, req *datastore.ListAttestedNodesEventsRequest) (*datastore.ListAttestedNodesEventsResponse, error) {
+	if !fflag.IsSet(fflag.FlagEventsBasedCache) {
+		return &datastore.ListAttestedNodesEventsResponse{}, nil
+	}
+
+	var events []AttestedNodeEvent
+	if err := tx.Find(&events, "id > ?", req.GreaterThanEventID).Error; err != nil {
+		return nil, sqlError.Wrap(err)
+	}
+
+	resp := &datastore.ListAttestedNodesEventsResponse{}
+	for _, event := range events {
+		resp.SpiffeIDs = append(resp.SpiffeIDs, event.SpiffeID)
```

Comment:
considering the volume of events, i wonder if we should preallocate this slice to len(events)

Same would be applicable to the entry events listing code.

Reply:
Done